varcode/variant_annotator.py

```diff
@@ -40,14 +40,38 @@ def describe_variant(self, variant):
 
         # dictionary from gene ID to list of transcript effects
         gene_transcript_effects_groups = {}
+
+        # mapping from Transcript objects to errors encountered
+        # while trying to annotated them
+        errors = {}
+
         for (gene_id, transcripts) in overlapping_transcript_groups.items():
             effects = []
             for transcript in transcripts:
-                effect = infer_transcript_effect(variant, transcript)
-                effects.append(effect)
+                try:
+                    effects.append(infer_transcript_effect(variant, transcript))
+                except (AssertionError, ValueError) as error:
+                    if raise_on_error:
+                        raise
+                    else:
+                        logging.warn(
+                            "Encountered error annotating %s for %s: %s",
+                            variant,
+                            transcript,
+                            error)
+                    errors[transcript] = error
```

Comment:
When you error on an `infer_transcript_effect(variant, transcript)`, you collect all the errors into a map from transcript to error. Thoughts on putting the error in the transcript effect object itself, instead of a map from transcript to error in the variant effect object?